pkg/virt-operator/resource/apply/rbac.go

```diff
@@ -34,49 +34,50 @@ func (r *Reconciler) createOrUpdateRoleBinding(rb *rbacv1.RoleBinding, imageTag
 	return r.createOrUpdate(rb, imageTag, imageRegistry, id, true)
 }
 
-func (r *Reconciler) createOrUpdate(role runtime.Object,
+func (r *Reconciler) createOrUpdate(required runtime.Object,
 	imageTag, imageRegistry, id string,
 	avoidIfServiceAccount bool) (err error) {
 
-	roleTypeName := role.GetObjectKind().GroupVersionKind().Kind
-	createRole := r.getCreateFunction(role)
-	updateRole := r.getUpdateFunction(role)
+	roleTypeName := required.GetObjectKind().GroupVersionKind().Kind
 
-	cachedRoleInterface, exists, _ := r.getCache(role).Get(role)
-	roleMeta := getMetaObject(role)
-	if avoidIfServiceAccount && !r.stores.ServiceMonitorEnabled && (roleMeta.Name == rbac.MONITOR_SERVICEACCOUNT_NAME) {
+	cachedRoleInterface, exists, _ := r.getCache(required).Get(required)
+	requiredMeta := getMetaObject(required)
+	if avoidIfServiceAccount && !r.stores.ServiceMonitorEnabled && (requiredMeta.Name == rbac.MONITOR_SERVICEACCOUNT_NAME) {
 		return nil
 	}
 
-	injectOperatorMetadata(r.kv, roleMeta, imageTag, imageRegistry, id, true)
+	injectOperatorMetadata(r.kv, requiredMeta, imageTag, imageRegistry, id, true)
 	if !exists {
 		// Create non existent
-		err = createRole()
+		err = r.getCreateFunction(required)()
 		if err != nil {
-			return fmt.Errorf("unable to create %v %+v: %v", roleTypeName, role, err)
+			return fmt.Errorf("unable to create %v %+v: %v", roleTypeName, required, err)
 		}
-		log.Log.V(2).Infof("%v %v created", roleTypeName, roleMeta.GetName())
+		log.Log.V(2).Infof("%v %v created", roleTypeName, requiredMeta.GetName())
 		return nil
 	}
 
-	modified := resourcemerge.BoolPtr(false)
-	cachedRole := cachedRoleInterface.(runtime.Object)
-	cachedRoleMeta := getMetaObject(cachedRole)
-	resourcemerge.EnsureObjectMeta(modified, cachedRoleMeta.DeepCopy(), *roleMeta)
-	enforceAPIGroup(cachedRole, role)
+	metaChanged := resourcemerge.BoolPtr(false)
+	existingCopy := cachedRoleInterface.(runtime.Object).DeepCopyObject()
+	existingCopyMeta := getMetaObject(existingCopy)
 
-	// there was no change to metadata, the generation matched
-	if !*modified && arePolicyRulesEqual(role, cachedRole) {
-		log.Log.V(4).Infof("%v %v already exists", roleTypeName, roleMeta.GetName())
+	resourcemerge.EnsureObjectMeta(metaChanged, existingCopyMeta, *requiredMeta)
+	enforceAPIGroup(existingCopy, required)
+
+	specChanged := changeExistingByRequired(existingCopy, required)
+
+	if !*metaChanged && !specChanged {
+		log.Log.V(4).Infof("%v %v already exists", roleTypeName, requiredMeta.GetName())
 		return nil
 	}
 
 	// Update existing, we don't need to patch for rbac rules.
-	err = updateRole()
+	updateFunc := r.getUpdateFunction(existingCopy)
```

Comment:
if we are going to do this, lets just do...
`err = r.getUpdateFunction(existingCopy)()`